traitlets/traitlets.py

```diff
@@ -720,12 +720,29 @@ def tag(self, **metadata):
 
         >>> Int(0).tag(config=True, sync=True)
         """
-        maybe_constructor_keywords = set(metadata.keys()).intersection({'help','allow_none', 'read_only', 'default_value'})
+        maybe_constructor_keywords = set(metadata.keys()).intersection(
+            {"help", "allow_none", "read_only", "default_value"}
+        )
         if maybe_constructor_keywords:
             warn('The following attributes are set in using `tag`, but seem to be constructor keywords arguments: %s '%
                     maybe_constructor_keywords, UserWarning, stacklevel=2)
 
         self.metadata.update(metadata)
+
+        # allow from_string to be overridden via metadata
+        if self.metadata.get("from_string"):
+            self.from_string = self.metadata["from_string"]
```

Comment:
Why not look this up in the `metadata` dict rather than assign to an attr and find it there?

Reply:
Nevermind, I guess `from_string_list` works the same way.

Reply:
Yeah, it's the TraitType subclass priority I was thinking of. The alternative was to implement `_from_string` private method that does the lookup, but I thought that might get confusing.